pkg/controller/vitesskeyspace/reconcile_resharding.go

```diff
@@ -120,8 +122,26 @@ func (r *reconcileHandler) reconcileResharding(ctx context.Context) (reconcile.R
 		sort.Strings(errorMsgs)
 		r.setConditionStatus(planetscalev2.VitessKeyspaceReshardingInSync, corev1.ConditionFalse, "Error", fmt.Sprintf("VReplication reported an error: %v", errorMsgs[0]))
 	case planetscalev2.WorkflowCopying:
+		// Aggregate tablet maps for all source shards.
+		sourceRowCount, err := r.shardsRowCount(ctx, workflowStatus.SourceShards)
+		if err != nil {
+			r.recorder.Eventf(r.vtk, corev1.EventTypeWarning, "SourceShardsRowCountFailed", "failed to aggregate row count for source shards: %v", err)
+			return resultBuilder.RequeueAfter(topoRequeueDelay)
+		}
+		targetRowCount, err := r.shardsRowCount(ctx, workflowStatus.TargetShards)
+		if err != nil {
+			r.recorder.Eventf(r.vtk, corev1.EventTypeWarning, "TargetShardsRowCountFailed", "failed to aggregate row count for target shards: %v", err)
+			return resultBuilder.RequeueAfter(topoRequeueDelay)
+		}
+		percentComplete := int(math.Floor((float64(targetRowCount) / float64(sourceRowCount)) * 100.0))
```

Comment:
> we could be losing information here potentially as we go from uint64 to float64. Do we think this could be problematic?

I don't think it's a problem. Row counts up to 2^53 - 1 will be represented exactly in a float64, and even above that, the error as a percentage will almost never be enough to change the truncated result.

> Should we just rely on floor division instead?

Integer arithmetic would work too, although it technically risks overflow in the unlikely event that the row count is close to 2^64, since you'd have to multiply by 100 before dividing.

My only suggestion here is that there's no need to call `Floor()` if we're about to convert to an integer anyway:

```suggestion
		percentComplete := int(float64(targetRowCount) / float64(sourceRowCount) * 100.0)
```

Reply:
Done.